### PR TITLE
Fixed 'to-do' item in birthday validation

### PR DESCRIPTION
**Pull request by itsjoekent**
- Supports leap years
- Last day of month is no longer just 31 & is based off actual last day in months 
- Prevents people from using 0 as a month

Hopefully my first contribution to the site? Yay?!

### CODE REVIEW

**Comment by barryclark** [comment]
@deadlybutter when opening a pull request, @ mention whoever should review it (and anyone else that it's important to see the code). For a fendery thing like this I'd @DFurnes @tjrosario @weerd.

**Comment by itsjoekent** [comment]
alrighty! I already poked @weerd about it in hipchat but ill do this in the future

**Comment by barryclark** [comment]
Sweet. First dosomething/dosomething PR -- woo woo! :dancers:

File: lib/themes/dosomething/paraneue_dosomething/js/validation/auth.js [hunk 1]
```diff
@@ -64,16 +64,21 @@ define(function(require) {
...
-    if (birthMonth > 12) {
+    if (birthMonth > 12 || birthMonth == 0) {
       return done({
         success: false,
         message: Drupal.t("That doesn't seem right.")
       });
     }
 
+    //list of last days in months and check for leap year
+    var endDates = [0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
```
Comment (weerd) [inline lib/themes/dosomething/paraneue_dosomething/js/validation/auth.js:75]: just to help clarify, why does the endDates array start with `0`?

Comment (itsjoekent) [inline lib/themes/dosomething/paraneue_dosomething/js/validation/auth.js:75]: So I can automatically plug the month into the array (so [1] refers to January, which is 31 days long) - I could subtract, but I already know the value is between 1-12 so I figured this made more sense

Comment (itsjoekent) [inline lib/themes/dosomething/paraneue_dosomething/js/validation/auth.js:75]: line 81 shows what I mean

Comment (weerd) [inline lib/themes/dosomething/paraneue_dosomething/js/validation/auth.js:75]: Cool. Yeah, the whole months array starting at index `0` and accounting for subtracting `1` always got to me. @DFurnes might think different, but looks cool to me :tropical_drink:

**Comment by weerd** [comment]
:+1: Looks snazzy!

**Comment by aaronschachter** [comment]
was there an issue for this, @deadlybutter ?

**Comment by itsjoekent** [comment]
I couldn't find one, was just a TO-DO in the comments

**Comment by DFurnes** [comment]
:thumbsup:

**Pull request merged**